templates/Dockerfile.template

```diff
@@ -21,7 +31,21 @@ RUN set -x \
 	&& wget -q -O /usr/local/src/gosu.tar.gz "https://github.com/tianon/gosu/archive/$GOSU_VERSION.tar.gz" \
 	# extract gosu binary and check signature
 	&& export GNUPGHOME="$(mktemp -d)" \
-	&& gpg --keyserver ha.pool.sks-keyservers.net --recv-keys {{ $gosu.Gpg }} \
+	&& found='' \
+	&& for server in \
+		pool.sks-keyservers.net \
+		ha.pool.sks-keyservers.net \
+		pgp.mit.edu \
+		na.pool.sks-keyservers.net \
+		eu.pool.sks-keyservers.net \
+		oc.pool.sks-keyservers.net \
+		ha.pool.sks-keyservers.net \
+		hkp://p80.pool.sks-keyservers.net:80 \
+		hkp://keyserver.ubuntu.com:80 \
+	; do \
+		gpg --keyserver $server --recv-keys {{ $gosu.Gpg }} \
+			&& found="yes" && break; \
+	done; test -n "$found" \
```

Comment:
We're mixing `;` and `&&` and I don't know how that behaves. Shouldn't we use `&&` here as well?

Reply:
We shouldn't.

We want to break a loop in case of success of gpg download, so we use `&&` internally.
After a loop, we want to test for it, and to be sure that test will be executed.

If we use &&, then test will not be executed if all gpg calls fails.

For example:
```
#!/bin/bash
echo "FIRST"
for i in 1 ; do
  false && break
done; echo $?
echo "SECOND"
for i in 1 ; do
  false && break
done && echo $?
```
will output
```
FIRST
1
SECOND
```